.changeset/metal-colts-begin.md

```diff
@@ -0,0 +1,5 @@
+---
+"@apollo/gateway": minor
+---
+
+feat: onSchema will be used new hook implementation
```

Comment:
should be "add `onSchemaWillBeUsed` hook. Consumers can use this new hook to trigger custom logic just before a new schema is passed to the Gateway executor`